policy-controller/k8s/index/src/inbound/routes.rs

```diff
@@ -1,24 +1,29 @@
+use crate::routes;
 use ahash::AHashMap as HashMap;
 use anyhow::{bail, Error, Result};
-use k8s_gateway_api as api;
 use linkerd_policy_controller_core::{
-    inbound::{Filter, HttpRoute, HttpRouteRule},
+    inbound::{Filter, InboundRoute, InboundRouteRule},
     routes::{HttpRouteMatch, Method},
     POLICY_CONTROLLER_NAME,
 };
 use linkerd_policy_controller_k8s_api::{
-    self as k8s, gateway, gateway as k8s_gateway_api,
+    self as k8s, gateway,
     policy::{httproute as policy, Server},
 };
 use std::fmt;
 
 #[derive(Clone, Debug, Eq, PartialEq)]
-pub struct RouteBinding {
+pub struct RouteBinding<MatchType> {
     pub parents: Vec<ParentRef>,
-    pub route: HttpRoute,
+    pub route: InboundRoute<MatchType>,
     pub statuses: Vec<Status>,
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum TypedRouteBinding {
```

Comment:
Given that we don't want to mix and match GrpcRoutes and HttpRoutes in the same map, I don't think we need a enum type like this.